thrift/zipkinCore.thrift

```diff
@@ -198,6 +197,17 @@ const string HTTP_RESPONSE_SIZE = "http.response.size"
  */
 const string LOCAL_COMPONENT = "lc"
 
+#***** Annotation.value or BinaryAnnotation.key ******
+/**
+ * When an annotation value, this indicates when an error occurred. When a
+ * binary annotation key, the value is a human readable message associated
+ * with an error.
+ *
+ * Note that RPC spans often include both client and server hosts. It is
+ * possible that only one side perceived the error.
+ */
+const string ERROR = "error"
```

Comment:
I would suggest clarifying the difference between the two, i.e. annotation (event) means some error occurred in the span (at a certain time) but not necessarily critical, while binary annotation (tag) means the application considers the operation represented by the span to have failed. For example, if a request failed on first try and succeeded on a retry, I would expect an event after the first failure, but no error tag on the overall span.

These don't have to be strict guidelines, but at least promote some consistency in instrumentation.